src/entities/network-config.js

```diff
@@ -48,10 +56,21 @@ class NetworkConfigImp implements NetworkConfig {
     this.#networkMagic = network.networkMagic
     this.#networkProtocol = network.protocol
     this.#networkDiscrimination = network.networkDiscrimination
+    this.#dataProvider = network.dataProvider
   }
 
   networkName = () => this.#networkName
 
+  dataProvider = () => {
```

Comment:
be careful defining member variables like this without specifying the type. Flow may not automatically carry it over to other modules. It's better to to
```
dataProvider: void => void = () => {
}
```